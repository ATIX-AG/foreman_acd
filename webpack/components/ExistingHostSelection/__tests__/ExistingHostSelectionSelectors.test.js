import { testSelectorsSnapshotWithFixtures } from 'react-redux-test-utils';

import {
  selectServiceId,
  selectAvailableHosts,
  selectAlreadyUsedHosts,
  selectSelectedHosts,
  selectHostsInHostgroup,
} from '../ExistingHostSelectionSelectors';

import { existingHostSelectionConfData1 } from '../__fixtures__/existingHostSelectionConfData1.fixtures';

const stateFactory = obj => ({
  foremanAcd: {
    existingHostSelectionConf: obj,
  },
});

const fixtures = {
  'should return serviceId from existingHostSelectionConfData1 fixtures': () =>
    selectServiceId(stateFactory(existingHostSelectionConfData1)),
  'should return availableHosts from existingHostSelectionConfData1 fixtures': () =>
    selectAvailableHosts(stateFactory(existingHostSelectionConfData1)),
  'should return alreadyUsedHosts from existingHostSelectionConfData1 fixtures': () =>
    selectAlreadyUsedHosts(stateFactory(existingHostSelectionConfData1)),
  'should return selectedHosts from existingHostSelectionConfData1 fixtures': () =>
    selectSelectedHosts(stateFactory(existingHostSelectionConfData1)),
  'should return hostsInHostgroup from existingHostSelectionConfData1 fixtures': () =>
    selectHostsInHostgroup(stateFactory(existingHostSelectionConfData1)),
};

describe('ExistingHostSelectionSelectors', () =>
  testSelectorsSnapshotWithFixtures(fixtures));
