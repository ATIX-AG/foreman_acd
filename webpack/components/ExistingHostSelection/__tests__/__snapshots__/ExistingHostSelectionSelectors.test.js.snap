// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExistingHostSelectionSelectors should return alreadyUsedHosts from existingHostSelectionConfData1 fixtures 1`] = `
Array [
  Object {
    "disabled": true,
    "label": "isaac-anselm",
    "tooltipText": "Host already used for this application instance",
    "value": "isaac-anselm",
  },
]
`;

exports[`ExistingHostSelectionSelectors should return availableHosts from existingHostSelectionConfData1 fixtures 1`] = `
Array [
  Object {
    "label": "lllllll",
    "value": "lllllll",
  },
]
`;

exports[`ExistingHostSelectionSelectors should return hostsInHostgroup from existingHostSelectionConfData1 fixtures 1`] = `undefined`;

exports[`ExistingHostSelectionSelectors should return selectedHosts from existingHostSelectionConfData1 fixtures 1`] = `Array []`;

exports[`ExistingHostSelectionSelectors should return serviceId from existingHostSelectionConfData1 fixtures 1`] = `2`;
